Move image build to proper temp dir

johnsca opened the pull request:
Jenkins was running into a strange issue cleaning up the build.tmp directory, so this creates a proper temp dir and does the building in that instead.

Cynerva commented:
This silently fails:

```
$ ./build
[+] Building 1.0s (8/8) FINISHED
 => [internal] load .dockerignore                                                                                                                                                                                                                                           0.1s
 => => transferring context: 2B                                                                                                                                                                                                                                             0.0s
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                        0.1s
 => => transferring dockerfile: 38B                                                                                                                                                                                                                                         0.0s
 => [internal] load metadata for docker.io/library/ubuntu:20.04                                                                                                                                                                                                             0.0s
 => [internal] load build context                                                                                                                                                                                                                                           0.0s
 => => transferring context: 42B                                                                                                                                                                                                                                            0.0s
 => CACHED [1/3] FROM docker.io/library/ubuntu:20.04                                                                                                                                                                                                                        0.0s
 => [2/3] RUN git clone https://github.com/intel/sriov-cni --branch v2.3 --depth 1;     cd sriov-cni;     make;     mkdir -p /opt/cni/bin;     cp sriov-cni/build/sriov /opt/cni/bin/sriov;     rm -rf sriov-cni                                                            0.6s
 => [3/3] COPY install-sriov-cni.sh /install-sriov-cni.sh                                                                                                                                                                                                                   0.1s
 => exporting to image                                                                                                                                                                                                                                                      0.1s
 => => exporting layers                                                                                                                                                                                                                                                     0.1s
 => => writing image sha256:830eac7b6c23cde6c4e0820940cbc497f0517ec1a6c17976e0882cc58548ce7f                                                                                                                                                                                0.0s
 => => naming to docker.io/library/sriov-cni
```

Notice how the RUN command only took 0.6s? That's not right. For reference, it takes me ~2 seconds to clone the repository, and ~13 seconds to run `make`. I can confirm that /opt/cni/bin/sriov is missing from the built image:

```
$ docker run -it --entrypoint /bin/bash sriov-cni
root@ead17a2a1b66:/# ls /opt/cni/bin
root@ead17a2a1b66:/#
```

I'm pretty sure that the ubuntu:20.04 image does not include git or make. You would need to install them with apt. Adding those will bloat the image. I'm not a fan of this approach.

johnsca committed:
Move image build to proper temp dir

Jenkins was running into a strange issue cleaning up the build.tmp
directory, so this creates a proper temp dir and does the building in
that instead.

johnsca commented on image/build:
```diff
@@ -1,12 +1,14 @@
+tmpdir="$(mktemp -d)"
+trap 'rm -rf "$tmpdir"' EXIT
+(
+ cp Dockerfile install-sriov-cni.sh "$tmpdir"
```
FWIW, I originally tried passing in the tmpdir path to the built sriov binary as a `--build-arg` but Docker got all snitty about the path, so I had to just do everything in the tmpdir instead.

Specifically:

```
 => => transferring context: 42B                                                                                                                                                                                                                                              0.0s
 => ERROR [2/3] COPY /tmp/tmp.zkeZ4XpaDY/build/sriov /opt/cni/bin/sriov                                                                                                                                                                                                       0.0s
------
 > [2/3] COPY /tmp/tmp.zkeZ4XpaDY/build/sriov /opt/cni/bin/sriov:
------
failed to solve with frontend dockerfile.v0: failed to build LLB: failed to compute cache key: "/tmp/tmp.zkeZ4XpaDY/build/sriov" not found: not found
```

Cynerva commented:
Works for me.
